### PR TITLE
[release/9.16.4] BUG AB#37522 - Fixed zIndexes

### PR DESCRIPTION
**Pull request by morethanfire**
Updated `zIndex` in order to stop the bugs that keep popping up time-to-time.

Components updated:
- `./src/inputs/datePickerInput`
- `./src/inputs/prompt`
- `./src/surfaces/drawer`
- `./src/surfaces/drawerDeprecated` (I know...)
- `./src/surfaces/modal`

### CODE REVIEW

File: src/surfaces/modal/modal.jsx [hunk 1]
```diff
@@ -242,11 +137,26 @@ class Modal extends React.Component {
...
-                this._onOpen();
+                this.onOpen();
             });
         }
     }
 
+    componentDidUpdate(prevProps, prevState) {
+        if (!prevProps.isOpen && this.props.isOpen) {
+            this.setState({
+                isOpen: this.props.isOpen,
```
Comment (groberts314) [inline src/surfaces/modal/modal.jsx:148]: Is it worth destructuring from `this.props` here to appease ESLINT?

**Review by groberts314 — APPROVED** [review]
Looks pretty good.  I flagged one NAB if you're up for it.  We're kinda under the gun, we can let it be too and just :shipit:

File: src/surfaces/modal/modal.jsx (hunk 1 above)
Comment (morethanfire) [reply src/surfaces/modal/modal.jsx:148]: Probably, but i already did this in v.10. I am wondering if we can get some of that work into v.9 one day.

Comment (morethanfire) [reply src/surfaces/modal/modal.jsx]: Oh, nm, I realized this was the one component i tried to get the red out of my face. I destructured.

Comment (groberts314) [reply src/surfaces/modal/modal.jsx]: Lookin' sharp.  Resolving.

**Comment by groberts314** [comment]
Definitely :shipit: now 😄 
You got the release and NPM publish?  Or you would you like me to assist?

**Review by sircris — APPROVED** [review]
Lots of code changes but nothing seems concerning. I need to test this out to have confidence with this.

**Pull request merged**